Blogroll: Update blog appender styling and functionality

agrullon95 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes https://github.com/Automattic/wp-calypso/issues/82173

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Updates blog appender edge case based on design feedback

Replaces blog appender placeholder site icon
Updates message if no results are found in blog appender

<img width="293" alt="CleanShot 2023-09-26 at 23 00 01@2x" src="https://github.com/Automattic/jetpack/assets/10482592/98753eb4-4f48-4624-a593-e4c57be688b0">

<img width="287" alt="CleanShot 2023-09-26 at 23 00 49@2x" src="https://github.com/Automattic/jetpack/assets/10482592/b2e6ad83-d094-4367-98b7-0ec383db11bc">

Designs: aOWo8Hpzper3EIXwPbuyzU-fi-361_3369


### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Checkout branch
* Navigate to the editor and add blogroll block
* Open the blogroll appenderVerify placeholder site icon has been updated to match designs
* Open the blogroll appender, search for a random site, and verify the "No results" message matches the designs

agrullon95 committed:
Update "no results" message for blog appender

agrullon95 committed:
Update placeholder site icon

agrullon95 committed:
changelog

heavyweight commented on projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender-results/style.scss:
```diff
@@ -48,7 +72,7 @@
 				overflow: hidden;
 
-				> * {
+				>* {
```
sm0l formatting issue

heavyweight commented on projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender/index.js:
```diff
@@ -45,6 +45,7 @@ export default function BlogrollAppender( { subscriptions, clientId } ) {
 						showPlaceholder={ ! searchInput.trim() }
```
Maybe we can remove this input now, since we are passing the input value below

alshakero committed:
Address feedback

alshakero reviewed (dismissed):
I rebased and fixed conflicts. I tested and all works.

I do have to note that the default icon wasn't changed on the frontend. I don't know if that's intentional.

agrullon95 commented:
> I rebased and fixed conflicts. I tested and all works.
> 
> I do have to note that the default icon wasn't changed on the frontend. I don't know if that's intentional.

Yes, I intentionally only updated the icon for the blog appender. It will show up on the frontend correctly for a site selected from the blog appender, but I still need to replace the placeholder site icon in the other areas of the block (such as the blogroll recommendations).

agrullon95 committed:
Merge branch 'trunk' into update/blogroll-appender-edge-cases-i1

heavyweight committed:
Merge branch 'trunk' into update/blogroll-appender-edge-cases-i1

heavyweight commented on projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender-results/placeholder-site-icon.svg:
Ideally, we would use the icon component for a site icon, but this would not render on the front-end.
So having it as svg is fine.
I would prefer to have a plain black and white svg without any customizations, and customize that with css to get what we want

alshakero committed:
Merge branch 'trunk' into update/blogroll-appender-edge-cases-i1

agrullon95 committed:
Merge branch 'trunk' into update/blogroll-appender-edge-cases-i1

agrullon95 committed:
Merge branch 'trunk' into update/blogroll-appender-edge-cases-i1

agrullon95 committed:
Update placeholder svg usage

alshakero commented on projects/plugins/jetpack/extensions/blocks/blogroll/blogroll-item/blogroll-item.php:
```diff
@@ -74,7 +74,7 @@ function load_assets( $attr, $content, $block ) {
 
 	if ( empty( $icon ) ) {
-		$icon = 'https://s0.wp.com/i/webclip.png';
+		$icon = "data:image/svg+xml,%3Csvg width='48' height='48' viewBox='0 0 48 48' fill='none' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'%3E%3Crect width='48' height='48' fill='%23E5E5E5'/%3E%3Cg id='edge cases'%3E%3Cpath d='M-5746 -1273C-5746 -1274.1 -5745.1 -1275 -5744 -1275H2020C2021.1 -1275 2022 -1274.1 2022 -1273V635C2022 636.105 2021.1 637 2020 637H-5744C-574 ...
```
We can add a file and use `plugin_url` to get the path. Base64 is ~67% as efficient as binary files. And it's noisy.

alshakero reviewed (commented):
Non-blocker

agrullon95 committed:
Remove placeholder site icon in createBlockFromRecommendation

agrullon95 committed:
Update placeholder-site-icon svg

agrullon95 committed:
Update logic to handle placeholder site icon

agrullon95 committed:
Merge branch 'trunk' into update/blogroll-appender-edge-cases-i1

alshakero commented on projects/plugins/jetpack/extensions/blocks/blogroll/blogroll-item/editor.scss:
```diff
@@ -6,9 +6,21 @@
+			&.empty-site-icon {
+				background: url('../placeholder-site-icon.svg');
+				background-size: cover;
+				border-radius: 50%;
```
I guess we don't need border-radius, the SVG is already round and transparent.

alshakero commented on projects/plugins/jetpack/extensions/blocks/blogroll/style.scss:
```diff
@@ -41,6 +41,16 @@
+				&.empty-site-icon {
+					background: url('./placeholder-site-icon.svg');
+					background-size: cover;
+					border-radius: 50%;
```
+1

alshakero reviewed (dismissed):
LGTM

heavyweight committed:
BlogrollItem: remove icon do_blocks

heavyweight committed:
BlogrollItem: minor placeholder fixes

heavyweight committed:
BlogrollItem: minor placeholder fill fixes

heavyweight committed:
Merge branch 'trunk' into update/blogroll-appender-edge-cases-i1
